src/bench/locustfile.py

```diff
@@ -0,0 +1,219 @@
+    return db.to_dict(orient="records")
+
+
+class BaseAPIUser(FastHttpUser):
+    """Base API user."""
+
+    abstract: bool = True
+    username: str
+    password: str
+    counter: dict = {
+        "static": {"create": STATIC_DB_OFFSET, "bulk": 10000},
+        "status": 0,
+        "session": 0,
+    }
+
+    def on_start(self):
+        """Log user in on start."""
+        credentials = {
+            "username": self.username,
+            "password": self.password,
+        }
+        response = requests.post(f"{self.host}/auth/token", data=credentials)
+        token = response.json()["access_token"]
+        self.client.auth_header = f"Bearer {token}"
+        self.statique_db = load_statique_db(
+            Path(os.environ["QUALICHARGE_STATIQUE_DATA_PATH"])
+        )
+
+    @task
+    def whoami(self):
+        """Test the /whoami endpoint."""
+        self.client.get("/auth/whoami")
+
+    @task
+    def statique_list(self):
+        """Assess the /statique GET endpoint."""
+        with self.rest("GET", "/statique/?limit=100") as response:
+            if response.js["size"] == 0:
+                response.failure("Database contains no statique entries")
+        with self.rest("GET", "/statique/?limit=10") as response:
+            if response.js["size"] == 0:
+                response.failure("Database contains no statique entries")
+
+    @task
+    def statique_get(self):
+        """Assess the /statique/{id_pdc_itinerance} GET endpoint."""
+        id_pdc_itinerance = "FRALLEGO002006P3"
+        with self.rest("GET", f"/statique/{id_pdc_itinerance}") as response:
+            if "id_pdc_itinerance" not in response.js:
+                response.failure(
+                    f"Database does not contain target statique entry {id_pdc_itinerance}"
+                )
+
+    @task
+    def statique_update(self):
+        """Assess the /statique/{id_pdc_itinerance} PUT endpoint."""
+        data = self.statique_db[22]
+        data["date_maj"] = "2024-11-21"
+        id_pdc_itinerance = data["id_pdc_itinerance"]
+
+        with self.rest("PUT", f"/statique/{id_pdc_itinerance}", json=data) as response:
+            if "id_pdc_itinerance" not in response.js:
+                response.failure(
+                    f"Database does not contain target statique entry {id_pdc_itinerance}"
+                )
+
+    @task
+    def statique_create(self):
+        """Assess the /statique/ POST endpoint."""
+        index = self.counter["static"]["create"]
+        data = self.statique_db[index]
+
+        with self.rest("POST", "/statique/", json=data) as response:
+            if response.status_code == 200 and response.js["size"] == 0:
+                response.failure("No Statique entry was created")
+            self.counter["static"]["create"] += 1
+
+    @task
+    def statique_bulk(self):
+        """Assess the /statique/bulk POST endpoint."""
+        start = self.counter["static"]["bulk"]
+        limit = 50
```

Comment:
`limit` could be added as a parameter

Reply:
Same philosophy: things are not designed to be flexible but reproducible.